File: src/openprecedent/schemas/artifact.py
from enum import StrEnum

from pydantic import BaseModel, ConfigDict


class ArtifactType(StrEnum):
    FILE = "file"
    COMMAND_OUTPUT = "command_output"
    MESSAGE = "message"
    REPORT = "report"
    PATCH = "patch"


class Artifact(BaseModel):
    model_config = ConfigDict(extra="forbid")

    artifact_id: str
    case_id: str
    artifact_type: ArtifactType
    uri_or_path: str
    summary: str | None = None

File: rust/openprecedent-contracts/src/artifact.rs
use serde::{Deserialize, Serialize};
use strum::{Display, EnumString};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize, Display, EnumString)]
#[serde(rename_all = "snake_case")]
#[strum(serialize_all = "snake_case")]
pub enum ArtifactType {
    File,
    CommandOutput,
    Message,
    Report,
    Patch,
}

#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub struct Artifact {
    pub artifact_id: String,
    pub case_id: String,
    pub artifact_type: ArtifactType,
    pub uri_or_path: String,
    pub summary: Option<String>,
}
